onnxruntime/core/providers/acl/acl_execution_provider.cc

```diff
@@ -0,0 +1,153 @@
+
+  kernel_registry.Register(BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kAclExecutionProvider, kOnnxDomain, 1, 8, float, GlobalAveragePool)>());
+  kernel_registry.Register(BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kAclExecutionProvider, kOnnxDomain, 1, 8, float, GlobalMaxPool)>());
+
+  // Opset 10
+  kernel_registry.Register(BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kAclExecutionProvider, kOnnxDomain, 10, 10, float, MaxPool)>());
+  kernel_registry.Register(BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kAclExecutionProvider, kOnnxDomain, 10, 10, float, AveragePool)>());
+}
+
+std::shared_ptr<KernelRegistry> GetAclKernelRegistry() {
+  std::shared_ptr<KernelRegistry> kernel_registry = std::make_shared<KernelRegistry>();
+  RegisterACLKernels(*kernel_registry);
+  return kernel_registry;
+}
+
+}  // namespace acl
+
+ACLExecutionProvider::ACLExecutionProvider(const ACLExecutionProviderInfo& /*info*/)
+      : IExecutionProvider{onnxruntime::kAclExecutionProvider} {
+
+
+  auto default_allocator_factory = [](int) {
+    auto memory_info = onnxruntime::make_unique<OrtMemoryInfo>(ACL, OrtAllocatorType::OrtDeviceAllocator);
+    return onnxruntime::make_unique<CPUAllocator>(std::move(memory_info));
+  };
+
+  DeviceAllocatorRegistrationInfo default_memory_info{
+    OrtMemTypeDefault,
+    std::move(default_allocator_factory),
+    std::numeric_limits<size_t>::max()
+  };
+
+  InsertAllocator(CreateAllocator(default_memory_info));
+
+
+  auto cpu_allocator_factory = [](int) {
+    auto memory_info = onnxruntime::make_unique<OrtMemoryInfo>(
+      ACL_CPU, OrtAllocatorType::OrtDeviceAllocator, OrtDevice(), 0, OrtMemTypeCPUOutput);
+    return onnxruntime::make_unique<CPUAllocator>(std::move(memory_info));
+  };
+
+  DeviceAllocatorRegistrationInfo cpu_memory_info{
+    OrtMemTypeCPUOutput,
+    std::move(cpu_allocator_factory),
+    std::numeric_limits<size_t>::max()
+  };
+
+  InsertAllocator(CreateAllocator(cpu_memory_info));
+}
+
+ACLExecutionProvider::~ACLExecutionProvider() {
+}
+
+std::shared_ptr<KernelRegistry> ACLExecutionProvider::GetKernelRegistry() const {
+  static std::shared_ptr<KernelRegistry> kernel_registry = onnxruntime::acl::GetAclKernelRegistry();
+  return kernel_registry;
+}
+
+std::vector<std::unique_ptr<ComputeCapability>>
+ACLExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph,
+                                    const std::vector<const KernelRegistry*>& kernel_registries) const {
+  std::vector<std::unique_ptr<ComputeCapability>>
+      result = IExecutionProvider::GetCapability(graph, kernel_registries);
+
+  for (auto& rule : fuse_rules_) {
+    rule(graph, result);
+  }
+  return result;
+}
+
+void ACLExecutionProvider::InsertFusedRules(FuseRuleFn rule) {
```

Comment:
>InsertFusedRules [](start = 27, length = 16)

what's this used for? Is it used to fuse some nodes together? It should be in optimizer if it's common no matter which provider it is.

Reply:
This is a left over from the first ACL EP skeleton.

Reply:
Fixed in v3